docs/options.md

```diff
@@ -420,33 +420,33 @@ the package is compatible with all versions of Python that it can build.
 !!! note
     Rather than using this option, it's recommended you set
     `project.requires-python` in `pyproject.toml` instead:
-    Example `pyproject.toml`:
 
-        [project]
-        requires-python = ">=3.6"
+    ```toml
+    [project]
+    requires-python = ">=3.6"
+    ```
 
-        # Aside - in pyproject.toml you should always specify minimal build
-        # system options, like this:
+    A few things to note:
 
+      - If your project didn't already have a `pyproject.toml`, be sure to
+        set the build system to your build backend.
+
+        ```toml
         [build-system]
         requires = ["setuptools>=42", "wheel"]
         build-backend = "setuptools.build_meta"
-
-
-    Currently, setuptools has not yet added support for reading this value from
-    pyproject.toml yet, and so does not copy it to Requires-Python in the wheel
-    metadata. This mechanism is used by pip to scan through older versions of
-    your package until it finds a release compatible with the current version
-    of Python compatible when installing, so it is an important value to set if
-    you plan to drop support for a version of Python in the future.
-
-    If you don't want to list this value twice, you can also use the setuptools
-    specific location in `setup.cfg` and cibuildwheel will detect it from
-    there. Example `setup.cfg`:
-
-        [options]
-        python_requires = ">=3.6"
-
+        ```
+
+      - If you didn't already have a `pyproject.toml` that had a `[project]`
+        table, you should migrate values from `setup.py` or `setup.cfg`, or
```

Comment:
> you should migrate

My only remark is that support for PEP 621 was just added last Friday, so there might be sharp edges left.

---

The following is just maintainer babbling 😅:

We can actively start encouraging people to migrate to it, however in setuptools we decided to keep the `experimental` status for a while to guarantee that we have the freedom to modify how we implement it before committing to an specific design.

Changes in interface will not occur in the `[project]` table (after all it is standardized), but if the users reach out for `[tool.setuptools]` there is no guarantee the fields there will stay unchanged between versions.

If in the following days we are reported that the releases seriously break important packages for the community we might also consider yanking it and backing off a bit while we re-implement any problematic bits.

Reply:
Thanks for reviewing !

> My only remark is that support for PEP 621 was just added last Friday, so there might be sharp edges left.

Hm. Yes okay I see. So what's the conservative route for users with a `[project]` table? Would listing options in `dynamic` be the safest way to go?

Reply:
If the users specify as dynamic the fields that they are providing via `setup.py`, that would be safe.

I think the parts of PEP 621 important to keep in mind are:

> Build back-ends MUST raise an error if the metadata specifies a field in dynamic but the build back-end was unable to provide the data for it.

> Build back-ends MUST raise an error if the metadata specifies a field statically as well as being listed in dynamic.

> If the metadata does not list a field in dynamic, then a build back-end CANNOT fill in